### PR TITLE
Admin plan module

### PR DESCRIPTION
**Pull request by sofiya02**

### CODE REVIEW

File: Block/Adminhtml/Plan/Edit/Form.php [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        );
+ 
+        $fieldset->addField(
+            'plan_desc',
+            'textarea',
+            ['name' => 'plan_desc', 'label' => __('Plan Description'), 'title' => __('Plan Description'), 'required' => true]
+        );
+        // $fieldset->addField(
+        //     'magento_product_id',
+        //     'text',
```
Comment (HancilSequeira) [inline Block/Adminhtml/Plan/Edit/Form.php:97]: Remove this commented code from every where if not required

File: Model/Source/ProductList.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                'value' => $category->getId(),
+            ];
+        }
+ 
+        return $options;
+    }
+}
+
+
+// class ProductList implements \Magento\Framework\Data\OptionSourceInterface
```
Comment (HancilSequeira) [inline Model/Source/ProductList.php:39]: remove this code

File: Model/Source/ProductList.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    ) {
+        $this->collectionFactory = $collectionFactory;
+    }
+ 
+    public function toOptionArray()
+    {
+        $options[] = ['label' => '-- Please Select --', 'value' => ''];
+        $collection = $this->collectionFactory->create()
+            ->addAttributeToSelect('*');
+            ///->addAttributeToFilter('is_active', '1');
```
Comment (HancilSequeira) [inline Model/Source/ProductList.php:25]: remove this code

**Review by HancilSequeira — CHANGES_REQUESTED** [review]
remove this file `view/.DS_Store` and all the commented code

File: Block/Adminhtml/Plan/Edit/Form.php (hunk 1 above)
Comment (sofiya02) [reply Block/Adminhtml/Plan/Edit/Form.php:97]: done

File: Model/Source/ProductList.php (hunk 2 above)
Comment (sofiya02) [reply Model/Source/ProductList.php]: done

File: Model/Source/ProductList.php (hunk 3 above)
Comment (sofiya02) [reply Model/Source/ProductList.php]: done

File: Ui/DataProvider/Plan/Collection.php [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    /**
+     * Override _initSelect to add custom columns
+     *
+     * @return void
+     */
+    protected function _initSelect()
+    {
+//     SELECT `main_table`.*, `second`.`value`, `third`.`attribute_id` FROM `razorpay_plans` AS `main_table`
+//  LEFT JOIN `catalog_product_entity_varchar` AS `second` ON main_table.magento_product_id = second.entity_id
+//  LEFT JOIN `eav_attribute` AS `third` ON third.attribute_id = second.attribute_id WHERE (third.attribute_code='name' and second.entity_id=main_table.magento_product_id)
```
Comment (HancilSequeira) [inline Ui/DataProvider/Plan/Collection.php:22]: Remove the commented the query and intend the file properly

**Review by HancilSequeira — CHANGES_REQUESTED** [review]

File: Controller/Adminhtml/Plan/MassEnable.php [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    }
+    /**
+     * Execute action
+     *
+     * @return \Magento\Backend\Model\View\Result\Redirect
+     * @throws \Magento\Framework\Exception\LocalizedException|\Exception
+     */
+    public function execute()
+    {
+        //$enableStatus= array('plan_status'=>'0');
```
Comment (HancilSequeira) [inline Controller/Adminhtml/Plan/MassEnable.php:41]: remove this line

File: Ui/Component/Listing/Column/PlanActions.php [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+                $item[$this->getData('name')]['edit'] = [
+                    'href' => $this->urlBuilder->getUrl(
+                        'subscribed/plan/edit',
+                        ['id' => $item['entity_id']]
+                    ),
+                    'label' => __('Edit'),
+                    'hidden' => false,
+                ];
+                // $item[$this->getData('name')]['delete'] = [
+                //     'href' => $this->urlBuilder->getUrl(
```
Comment (HancilSequeira) [inline Ui/Component/Listing/Column/PlanActions.php:60]: Remove this

File: Ui/DataProvider/Plan/Collection.php [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+class Collection extends SearchResult
+{
+    /**
+     * Override _initSelect to add custom columns
+     *
+     * @return void
+     */
+    protected function _initSelect()
+    {
+$this->addFilterToMap('entity_id', 'main_table.entity_id');
```
Comment (HancilSequeira) [inline Ui/DataProvider/Plan/Collection.php:20]: Indent this function properly

**Review by HancilSequeira — CHANGES_REQUESTED** [review]

File: Ui/DataProvider/Plan/Collection.php (hunk 4 above)
Comment (sofiya02) [reply Ui/DataProvider/Plan/Collection.php]: done

File: Controller/Adminhtml/Plan/MassEnable.php (hunk 5 above)
Comment (sofiya02) [reply Controller/Adminhtml/Plan/MassEnable.php:41]: done

File: Ui/Component/Listing/Column/PlanActions.php (hunk 6 above)
Comment (sofiya02) [reply Ui/Component/Listing/Column/PlanActions.php:60]: Done

File: Ui/DataProvider/Plan/Collection.php (hunk 7 above)
Comment (sofiya02) [reply Ui/DataProvider/Plan/Collection.php:20]: done

**Review by HancilSequeira — APPROVED** [review]
LGTM

**Pull request merged**